From 065557c5813a7a9e68aac62adeb2244d4fdcf9db Mon Sep 17 00:00:00 2001
From: "Christopher R. Palmer" <crpalmer@gmail.com>
Date: Mon, 16 May 2016 05:21:28 -0400
Subject: [PATCH 07/10] init: Allow devices to opt-out of fsck'ing on power off

On the zenfone2 the fsck code patch kills alls processes.  This causes
watchdogd to get killed which then causes the system to emergency reboot
instead of powering off.

Change-Id: Ibcc74825df6571caa70e2dde7766a72b7749581b
---
 init/Android.mk | 4 ++++
 init/reboot.cpp | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/init/Android.mk b/init/Android.mk
index 6d91fa19b..f44685c53 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -56,6 +56,10 @@ LOCAL_SRC_FILES:= \
     watchdogd.cpp \
     vendor_init.cpp
 
+ifneq ($(TARGET_INIT_UMOUNT_AND_FSCK_IS_UNSAFE),)
+LOCAL_CFLAGS += -DUMOUNT_AND_FSCK_IS_UNSAFE
+endif
+
 LOCAL_MODULE:= init
 LOCAL_C_INCLUDES += \
     system/core/mkbootimg
diff --git a/init/reboot.cpp b/init/reboot.cpp
index 5c7ddf185..5f7131e2e 100644
--- a/init/reboot.cpp
+++ b/init/reboot.cpp
@@ -134,6 +134,7 @@ class MountEntry {
 
 // Turn off backlight while we are performing power down cleanup activities.
 static void TurnOffBacklight() {
+#ifndef UMOUNT_AND_FSCK_IS_UNSAFE
     static constexpr char OFF[] = "0";
 
     android::base::WriteStringToFile(OFF, "/sys/class/leds/lcd-backlight/brightness");
@@ -153,6 +154,7 @@ static void TurnOffBacklight() {
         std::string fileName = StringPrintf("%s/%s/brightness", backlightDir, dp->d_name);
         android::base::WriteStringToFile(OFF, fileName);
     }
+#endif
 }
 
 static void ShutdownVold() {
@@ -477,7 +479,9 @@ bool HandlePowerctlMessage(const std::string& command) {
         if (cmd_params.size() == 2 && cmd_params[1] == "userrequested") {
             // The shutdown reason is PowerManager.SHUTDOWN_USER_REQUESTED.
             // Run fsck once the file system is remounted in read-only mode.
+#ifndef UMOUNT_AND_FSCK_IS_UNSAFE
             run_fsck = true;
+#endif
         }
     } else if (cmd_params[0] == "reboot") {
         cmd = ANDROID_RB_RESTART2;
-- 
2.17.1

